component/prometheus/exporter/postgres/postgres.go

```diff
@@ -0,0 +1,71 @@
+package postgres
+
+import (
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/prometheus/exporter"
+	"github.com/grafana/agent/pkg/flow/rivertypes"
+	"github.com/grafana/agent/pkg/integrations"
+	"github.com/grafana/agent/pkg/integrations/postgres_exporter"
+	config_util "github.com/prometheus/common/config"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "prometheus.exporter.postgres",
+		Args:    Arguments{},
+		Exports: exporter.Exports{},
+		Build:   exporter.New(createIntegration, "postgres"),
+	})
+}
+
+func createIntegration(opts component.Options, args component.Arguments) (integrations.Integration, error) {
+	cfg := args.(Arguments)
+	return cfg.Convert().NewIntegration(opts.Logger)
+}
+
+// DefaultArguments holds the default arguments for the prometheus.exporter.postgres
+// component.
+var DefaultArguments = Arguments{}
+
+// Arguments configures the prometheus.exporter.postgres component
+type Arguments struct {
+	// DataSourceNames to use to connect to Postgres. This is marked optional because it
+	// may also be supplied by the POSTGRES_EXPORTER_DATA_SOURCE_NAME env var,
+	// though it is not recommended to do so.
+	DataSourceNames []rivertypes.Secret `river:"data_source_names,attr,optional"`
```

Comment:
There is some validation being done [inside the integration](https://github.com/grafana/agent/blob/main/pkg/integrations/postgres_exporter/postgres_exporter.go#L117), however it turns out that if neither was set [this line](https://github.com/grafana/agent/blob/main/pkg/integrations/postgres_exporter/postgres_exporter.go#L111) would result in a slice like: `[]string{""}` (since `os.Getenv` returns `""` if the env var is not set, and `strings.Split("", ",")` returns `[]string{""}`), so it was not detecting this since that is a slice of length 1.

I have updated the PR with a fix for this, but would be happy to make that a separate PR, if needed.